### ISSUE TITLE
[Migrated] Add ASGI support

### ISSUE THREAD

**Issue by jneves** [zappa/Zappa#898]
Originally from: https://github.com/Miserlou/Zappa/issues/2159 by [simonw](https://github.com/simonw)

Zappa currently only supports WSGI.

[ASGI](https://asgi.readthedocs.io/en/latest/) is a new standard for Python web programming with deep support for Python 3 asyncio.

If Zappa added support for ASGI, projects written using the following frameworks would all become deployable using Zappa:

- [Starlette](https://www.starlette.io/) and [FastAPI](https://fastapi.tiangolo.com/)
- [Sanic](https://sanic.readthedocs.io/en/latest/)
- [Quart](https://pgjones.gitlab.io/quart/)
- [Datasette](https://docs.datasette.io/en/stable/)

The [Mangum](https://github.com/jordaneremieff/mangum) project already implements an ASGI handler for Lambda.

The work by @delijati in this fork seems to indicate that modifying Zappa to support ASGI via Mangum could be achieved with a relatively small change: https://github.com/Miserlou/Zappa/compare/master...delijati:master

**Comment by delijati** [comment]
Please consider also my benchmark of **wsgi** vs **asgi** on lambda https://github.com/delijati/benchmark . I have no real time to look into the deeper cause but i assume it has to do with (Lambda cold start or so.) 

tldr;

WSGI: **Requests/sec:   4850.86**
ASGI: **Requests/sec:   4452.44**

**Comment by Corstiaan84** [comment]
Bumping to see is this will come back to life.

**Comment by delijati** [comment]
There is no real benefit for using ASGI with AWS Lambda but Zappa adaption could benefit from it.

**Comment by nigeljames-tess** [comment]
No real benefit? @delijati Pray tell, I would like to hear more.

**Comment by delijati** [comment]
> No real benefit? [@delijati](https://github.com/delijati) Pray tell, I would like to hear more.

https://stackoverflow.com/a/73397171 tldr; Lambda is a **sync** if you need ASGI use magnum
